mlir/lib/Quantum/Transforms/emit_catalyst_pyface.cpp

```diff
@@ -131,6 +121,17 @@ void wrapResultsAndArgsInTwoStructs(LLVM::LLVMFuncOp op, PatternRewriter &rewrit
     bool hasInputs = functionHasInputs(callee);
 
     LLVM::LLVMFunctionType functionType = op.getFunctionType();
+    Type ptrType = LLVM::LLVMPointerType::get(rewriter.getContext());
```

Comment:
```
!llvm.ptr<struct<(ptr<struct<(ptr<f64>, ptr<f64>, i64)>>, ptr<struct<(ptr<f64>, ptr<f64>, i64)>>)>>
```
things like that appear in the IR

Reply:
Can you take a look at the next file? One ending in `.ll` not in `.mlir`?

Reply:
For this specific file @erick-xanadu would know best if the patch makes sense, but the general approach in the PR seems correct to me 👍  We are just moving the "loss" of type information one level up, from the LLVM IR into the LLVM dialect. The only downside I see is tests might be a bit harder if they were specifically testing the type information.

Now looking at the code in a lot of places it can seem a bit redundant, like writing `charPtr = LLVM::LLVMPointerType()` and then writing it again for a different pointer, even though all pointers are the same, but it doesn't do any harm either.